### PR TITLE
Fixed #7439 : try-with-resource Grammer to close ActivityController after test

### LINKED ISSUE #7439: Call Robolectric.buildActivity in ShadowActivityTest with Java's try-with-resource grammer to close ActivityController after test
### Description

There are some direct callings of `Robolectric.buildActivity` in `ShadowActivityTest` without `try-with-resource` grammer to close `ActivityController` after usage, and the Android Studio shows some warnings for us. It's better if we can use `try-with-resource` to auto-close `ActivityController` after test.

### Steps to Reproduce

Using Android Studio to open Robolectric's `ShadowActivityTest`, and we can see above warnings.

### Robolectric & Android Version

Robolectric latest master branch.

### Link to a public git repo demonstrating the problem:

Robolectric itself.

### PR DESCRIPTION
**Pull request by arinmodi**
### Overview

Fixes : #7439

### Proposed Changes

- There are some direct callings of Robolectric.buildActivity in ShadowActivityTest without try-with-resource grammer to close ActivityController after usage, and the Android Studio shows some warnings for us. It's better if we can use try-with-resource to auto-close ActivityController after test.

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java [hunk 1]
```diff
@@ -118,23 +118,25 @@ public void shouldUseActivityLabelFromManifestAsTitleForActivityWithShortName()
...
   public void createActivity_noDisplayFinished_shouldFinishActivity() {
-    ActivityController<Activity> controller = Robolectric.buildActivity(Activity.class);
-    controller.get().setTheme(android.R.style.Theme_NoDisplay);
-    controller.create();
-    controller.get().finish();
-    controller.start().visible().resume();
-
-    activity = controller.get();
-    assertThat(activity.isFinishing()).isTrue();
+      try(ActivityController<Activity> controller = Robolectric.buildActivity(Activity.class)) {
```
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:121]: I think you need to run google-java-format locally: https://github.com/robolectric/robolectric/wiki/Running-google-java-format.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java [hunk 2]
```diff
@@ -118,23 +118,25 @@ public void shouldUseActivityLabelFromManifestAsTitleForActivityWithShortName()
...
   }
 
   @Config(minSdk = M)
   @Test
   public void createActivity_noDisplayNotFinished_shouldThrowIllegalStateException() {
     try {
-      ActivityController<Activity> controller = Robolectric.buildActivity(Activity.class);
-      controller.get().setTheme(android.R.style.Theme_NoDisplay);
-      controller.setup();
+      try(ActivityController<Activity> controller = Robolectric.buildActivity(Activity.class)) {
```
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:136]: There is an existing `try-catch` statement, we don't need to use a new one.

**Review by utzcoz — CHANGES_REQUESTED** [review]

**Comment by utzcoz** [comment]
@arinmodi Thanks for your contribution. There are some changes needed to be done. Please see comments of this PR.

**Comment by arinmodi** [comment]
@utzcoz, Thanks for review, i will do the changes as suggested

File: robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java (hunk 1 above)
Comment (arinmodi) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:121]: @utzcoz, can you please explain me the use of this google-java-format ?

Comment (arinmodi) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:121]: By the way, I have enabled the google-java-format plugin in the android studio, will that do ?

**Comment by utzcoz** [comment]
FYI @arinmodi could you squash commits to one? And it might need your later updates also keep one commit. Thanks.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java (hunk 1 above)
Comment (utzcoz) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:121]: You can follow above link to run google-java-format locally. We don't format all files of Robolectric and we only format changed lines/files.

Comment (arinmodi) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:121]: @utzcoz, what are these variables $HOME and $ROBOLECTRIC_ROOT, can I change that ?

**Comment by arinmodi** [comment]
@utzcoz, I am closing this and making another PR, it's just messed up

**Pull request closed without merging**